### PR TITLE
[feature] Add jdbc connection pooling

### LINKED ISSUE #36: Databse connection pooling
It would be beneficiary to the overall performance if we had database connection pools.

### PR DESCRIPTION
**Pull request by wallner**
This pull request uses the excellent [HikariCP](https://github.com/brettwooldridge/HikariCP)
JDBC connection pool. This fixes #36

### CODE REVIEW

**Comment by tkrille** [comment]
Integration tests are failing. Here's part of a stacktrace I found in the Tomcat's logs:

```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.zaxxer.hikari.HikariDataSource]: Constructor threw exception; nested exception is java.lang.ClassCastException: org.postgresql.Driver cannot be cast to javax.sql.DataSource
    at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:163)
    at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:122)
    at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:267)
    ... 46 more
Caused by: java.lang.ClassCastException: org.postgresql.Driver cannot be cast to javax.sql.DataSource
    at com.zaxxer.hikari.pool.PoolUtilities.initializeDataSource(PoolUtilities.java:102)
    at com.zaxxer.hikari.pool.BaseHikariPool.<init>(BaseHikariPool.java:156)
    at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:60)
    at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:48)
    at com.zaxxer.hikari.HikariDataSource.<init>(HikariDataSource.java:80)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
    at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:147)
    ... 48 more
```

**Comment by tkrille** [comment]
I suggest you use the more traditional approach by using `jdbcUrl` for configuring Hikari.

**Comment by tkrille** [comment]
fixes #36

**Comment by dacrome** [comment]
...or don't use the database driver class as datasource

**Comment by tkrille** [comment]
Making the data source configurable is not a viable option, because then we also have to put this in the configuration files.

**Comment by dacrome** [comment]
The data source do not has to be configurable because it is fix. But the reference of the data source bean should be passee instead of the classname, right?

**Comment by tkrille** [comment]
Can you squash the commits?

File: src/main/webapp/WEB-INF/jpa-configuration.xml [hunk 1]
```diff
@@ -50,13 +50,22 @@
...
-        <property name="url" value="${org.osiam.auth-server.db.url}"/>
-        <property name="username" value="${org.osiam.auth-server.db.username}"/>
-        <property name="password" value="${org.osiam.auth-server.db.password}"/>
+    <bean id="hikariConfig" class="com.zaxxer.hikari.HikariConfig">
+        <property name="poolName" value="springHikariCP" />
+        <property name="driverClassName" value="${org.osiam.auth-server.db.driver}" />
+        <property name="jdbcUrl" value="${org.osiam.auth-server.db.url}" />
+        <property name="dataSourceProperties">
+            <props>
+                <prop key="url">${org.osiam.auth-server.db.url}</prop>
```
Comment (tkrille) [inline src/main/webapp/WEB-INF/jpa-configuration.xml]: Is this still necessary? After all, you already specified the JDBC URL above.

**Comment by wallner** [comment]
squashed

File: src/main/webapp/WEB-INF/jpa-configuration.xml [hunk 2]
```diff
@@ -50,13 +50,17 @@
         </property>
     </bean>
 
-    <bean id="dataSource" class="org.springframework.jdbc.datasource.DriverManagerDataSource">
-        <property name="driverClassName" value="${org.osiam.auth-server.db.driver}"/>
-        <property name="url" value="${org.osiam.auth-server.db.url}"/>
-        <property name="username" value="${org.osiam.auth-server.db.username}"/>
-        <property name="password" value="${org.osiam.auth-server.db.password}"/>
+    <bean id="hikariConfig" class="com.zaxxer.hikari.HikariConfig">
+        <property name="poolName" value="springHikariCP" />
```
Comment (dacrome) [inline src/main/webapp/WEB-INF/jpa-configuration.xml]: I'm not sure if the pool name is logged or used somewhere (maybe later with metrics?), but using the name of an hikari spring example isn't that helpful. My suggestion: `osiam-auth-server-connection-pool`

Comment (dacrome) [inline src/main/webapp/WEB-INF/jpa-configuration.xml]: [Description of the poolName](https://github.com/brettwooldridge/HikariCP#frequently-used):

> This property represents a user-defined name for the connection pool and appears mainly in logging and JMX management consoles to identify pools and pool configurations. Default: auto-generated

Comment (tkrille) [inline src/main/webapp/WEB-INF/jpa-configuration.xml]: Yes, a better pool name would be great.

Comment (wallner) [inline src/main/webapp/WEB-INF/jpa-configuration.xml]: Done.

**Comment by tkrille** [comment]
LGTM now. Merged.

**Pull request merged**